index.js

```diff
@@ -2,5 +2,5 @@ export default function dlv(obj, key, def, p) {
 	p = 0;
 	key = key.split ? key.split('.') : key;
 	while (obj && p<key.length) obj = obj[key[p++]];
-	return obj===undefined ? def : obj;
-}
\ No newline at end of file
+	return (obj===undefined || p<key.length) ? def : obj;
```

Comment:
Could we not just have `obj == undefined` here and drop the strict equality check? :)

Reply:
I am not following, if we drop the strict equality check it will fail to return a defined key with a `null` value, and will return the default instead. For example:
```
var obj = { a: null }
delve(obj, "a", "foo")
=> "foo"
``` 
It should have returned `null`, it's a defined key. It should only return `undefined` when the key is not defined in the object.

The strict equality check is necessary to distinguish undefined keys from the defined keys with null values.
I hope this makes sense.

Reply:
Yeah you're right